ahds/header.py

```diff
@@ -8,27 +8,102 @@
 ::
 
     >>> from amira.header import AmiraHeader
-    >>> ah = AmiraHeader.from_file('file.am')
+    >>> ah = AmiraHeader.from_file('<somfile>.am')
     >>> print ah
 
 Each nested object is constructed from the :py:class:``Block`` class defined.
 
-There are four top-level attributes that every ``AmiraHeader`` will have:
+There are fife top-level attributes that every ``AmiraHeader`` will have:
 
 *    designation
 
 *    definitions
 
 *    parameters
 
+*    materials
+
 *    data_pointers
 
-Each attribute can be queried using the ``attrs`` attribute.
+All of them, exempt materials and parameters are marked deprecated and may be removed in
+future. The designation and defintions attributes just return the reference to the AmiraHeader
+instance which host all the attributes and data array declarations (definitions) found within
+the AmiraMesh and HyperSurface files. The data_pointers attribute collects from each
+attribute representing a valid array declaration the defined data definitions and returns the resulting
+list. New code should directly access the Block objects describing the available data directly
+from the corresponding array attribute of the AmiraHeader object. 
+
+The blocks defining the available data structures extend the basic Block to the corresponding 
+AmiraMeshDataStream and AmiraHxSurfaceDataStream blocks which allow to load and access the correspinding
+data using the data property defined by theses blocks. The additional stream_data, encoded_data and
+decoded_data attributes are deprecated and may be removed in future versions.
+
+The attrs attribute of all Blocks and extended Block objects including AmiraHeader it self
+which allows to query the attributes added during loading is deprecated and may be removed in future version.
+To query which attributes are defined on each block use the builtin dir function instead. To 
+access the defined attributes just use them liḱe any other static attribute.
 
+The following attributes are moved from the designation attribute to the basic header block:
+        
+        *    filetype e.g. ``AmiraMesh`` or ``HyperSurface``
+        *    dimensions e.g. ``3D``
+        *    format e.g. ``BINARY-LITTLE-ENDIAN``
+        *    version e.g. ``2.1``
+        *    extra_format (if present)
+        
 :: 
 
-    >>> print ah.data_pointers.attrs
-    ['data_pointer_1', 'data_pointer_2', 'data_pointer_3', 'data_pointer_4', 'data_pointer_5', 'data_pointer_6']
+    >>> print(ah.Nodes)
+    print(ah.Nodes)
+     +-Nodes: (<ahds.ahds_common.Block object at 0x7f5dc1a74d08>)
+     |  |  +-Coordinates: (<ahds.data_stream.AmiraMeshDataStream object at 0x7f5dc1a97458>)
+     |  |  |  +-block: 1
```

Comment:
Help me understand: is there are scenario where we can have more than one value to the `block` field? Is this redundant?

Reply:
yes on HyperSurface files Patches section can have multiple block which represent a single surface patch
The same for optional Surfaces Section where each block joins one or more patches into one surface. By this a single HyperSurface file could contain multiple surfaces consisting of multiple patches each.

For Coordinates there will only be one block for which code currently does not make a distinction as this would have to be handled explicitly. It can but would make things more complicated.